### DISCUSSION TITLE
Connect pgAdmin to PostgreSQL?

### DISCUSSION THREAD

**Discussion by russellgilbert** [teslamate-org/teslamate#2789] [Getting started]
I installed TeslaMate using Docker on a CentOS 7 server and everything appears to be running correctly and loading data via the Tesla API. I'd like to access the database via something similar to PHPMyAdmin (which I'm much more familiar with) and my understanding is that I should use pgAdmin (I'm familiar with Linux but I've never used PostgreSQL before).

I installed pgAdmin 3 on the same server but when I try to connect it to host "localhost" with port 5432, I get "could not connect to server: Connection refused". I also installed pgAdmin 4 on my Windows 11 machine, and opened TCP port 5432 in the CentOS firewall, but I get "Unable to connect to server: connection to server '192.168.1.12', port 5432 failed: Connection refused".

I also tried "netstat -tulpn" and "lsof -i -P -n" but I see no reference to 5432 or PostgreSQL.

When I edit postgresql.conf I see the following, which makes me think that it's already setup to listen on any address, using the default port 5432:

listen_addresses = '*'
port = 5432

I also tried explicitly setting: listen_addresses = 'localhost'.

Can I get some help accessing the database tables?

Thanks!
Russell

**Comment by DrMichael** [comment]
Have you opened 5432 in docker-compose.yml?

Like
```
    ports:
     - 5432:5432
```
in the database part?

**Reply by russellgilbert** [reply]
Thanks. I added those lines to the database section of TeslaMate's docker-compose.yml file, and then restarted the server, but everything still looks like I originally described it. No listening port open, and pgAdmin still can't connect.

Other ideas?

Thanks,
Russell

**Comment by russellgilbert** [comment]
Well, after a lot of shots in the dark, I finally got it listening on port 5432, according to netstat. I honestly don't know what caused it to suddenly start listening on that port, because when I tested it each time after a change, it failed. Then suddenly it started working, and continued working even after I backed out the most recent change. Confusing.

Now, when I try to connect pgAdmin 3 to the database, I get the error, "Error connecting to the server: SCRAM authentication requires libpq version 10 or above". When I do "locate libpq", I see that it's on version 5, but I don't see an easy way to upgrade it. I tried the suggestion of changing postgresql.conf and pg_hba.conf to use md5 but it didn't help.

However... When I tried pgAdmin 4 from the Windows 11 machine again, it worked! I had already opened port 5432 in the server's firewall, and I guess something about it being version 4 is what was needed.
